test/e2e/resize_nodes.go

```diff
@@ -560,6 +566,93 @@ var _ = Describe("Nodes [Disruptive]", func() {
 					}
 				}
 			})
+
+			// What happens in this test:
+			// 	Network connectivity on a node is cut off for 120 seconds to simulate network partition
+			// Expect to observe:
+			// 1. Node is marked NotReady after timeout by nodecontroller (40seconds)
+			// 2. All pods on node are marked NotReady shortly after #1
+			// 3. Node and pods return to Ready after connectivivty recovers
+			It("All pods on the unreachable node should be marked as NotReady upon the node turn NotReady "+
+				"AND all pods should be mark back to Ready when the node get back to Ready before pod eviction timeout", func() {
+				By("choose a node - we will block all network traffic on this node")
+				nodeOpts := api.ListOptions{}
+				nodes, err := c.Nodes().List(nodeOpts)
+				Expect(err).NotTo(HaveOccurred())
+				filterNodes(nodes, func(node api.Node) bool {
+					return isNodeConditionSetAsExpected(&node, api.NodeReady, true)
+				})
+				if len(nodes.Items) <= 0 {
+					Failf("No node were found Ready: %d", len(nodes.Items))
+				}
+				node := nodes.Items[0]
+				podOpts := api.ListOptions{FieldSelector: fields.OneTermEqualSelector(client.PodHost, node.Name)}
+				if err = waitForMatchPodsCondition(c, podOpts, "Running and Ready", podReadyTimeout, podRunningReady); err != nil {
+					Failf("Pods on node %s are not ready and running within %v: %v", node.Name, podReadyTimeout, err)
+				}
+				nodeOpts = api.ListOptions{FieldSelector: fields.OneTermEqualSelector("metadata.name", node.Name)}
+
+				By("Set up watch on node status")
+				w, err := c.Nodes().Watch(nodeOpts)
+				if err != nil {
+					Failf("Failed to set up watch: %v", err)
+				}
+
+				defer w.Stop()
+
+				By(fmt.Sprintf("Performing temporary network partition on node %s", node.Name))
+				partitionCmd := "nohup sh -c 'sleep 10 && sudo ifdown eth0 && sleep 120 && sudo ifup eth0' >/dev/null 2>&1 &"
+				Expect(issueSSHCommand(partitionCmd, testContext.Provider, &node)).NotTo(HaveOccurred())
+
+				By("Expect to observe node and pod status change from Ready to NotReady after network partition")
+				timer := time.After(nodeNotReadyTimeout)
+				notReady := false
+				timeout := false
+				var lastNode *api.Node
+				for !notReady && !timeout {
+					select {
+					case event, _ := <-w.ResultChan():
```

Comment:
With the informer you'd just have an event handler, something like https://github.com/kubernetes/kubernetes/blob/master/test/e2e/density.go#L362